src/frequenz/sdk/timeseries/_ringbuffer.py

```diff
@@ -422,10 +425,22 @@ def __setitem__(
             index_or_slice: Index or slice specification of the requested data.
             value: Value to set at the given position.
         """
-        self._buffer.__setitem__(index_or_slice, value)
+        # There seem to be 2 different mypy bugs at play here.
+        # First we need to check that the combination of input arguments are
+        # correct to make the type checker happy (I guess it could be inferred
+        # from the @overloads, but it's not currently working without this
+        # hack).
+        # Then we need to ignore a no-untyped-call error, for some reason it
+        # can't get the type for self._buffer.__setitem__()
+        if isinstance(index_or_slice, SupportsIndex) and isinstance(value, float):
+            self._buffer.__setitem__(index_or_slice, value)  # type: ignore[no-untyped-call]
+        elif isinstance(index_or_slice, slice) and isinstance(value, Iterable):
+            self._buffer.__setitem__(index_or_slice, value)  # type: ignore[no-untyped-call]
+        else:
+            raise TypeError("Incompatible input arguments")
```

Comment:
`TypeError` might need to be added in the docstring `Raises` session

Reply:
Same as above.